bin/new_element.dart

```diff
@@ -0,0 +1,193 @@
+      print('When creating elements in root directory of package, '
+      '-o <dir> must be specified');
+      exitCode = 1;
+      return;
+    }
+    outputDir = (new Directory('.')).resolveSymbolicLinksSync();
+  } else {
+    var outputDirLocation = new Directory(outputPath);
+    if (!outputDirLocation.existsSync()) {
+      outputDirLocation.createSync(recursive: true);
+    }
+    outputDir = (new Directory(outputPath)).resolveSymbolicLinksSync();
+  }
+
+  var pubspecDir = _findDirWithFile(outputDir, 'pubspec.yaml');
+
+  if (pubspecDir == null) {
+    print('Could not find pubspec.yaml when walking up from $outputDir');
+    exitCode = 1;
+    return;
+  }
+
+  var length = path.split(pubspecDir).length;
+  var distanceToPackageRoot = path.split(outputDir).length - length;
+
+  // See dartbug.com/20076 for the algorithm used here.
+  if (distanceToPackageRoot > 0) {
+    if (path.split(outputDir)[length] == 'lib') {
+      distanceToPackageRoot++;
+    } else {
+      distanceToPackageRoot--;
+    }
+  }
+
+  try {
+    _createBoilerPlate(
+        element, outputDir, distanceToPackageRoot);
+  } on Exception catch (e, t) {
+    print('Error creating files in $outputDir');
+    print('$e $t');
+    exitCode = 1;
+    return;
+  }
+
+  return;
+}
+
+String _findDirWithFile(String dir, String filename) {
+  while (!new File(path.join(dir, filename)).existsSync()) {
+    var parentDir = path.dirname(dir);
+    // If we reached root and failed to find it, bail.
+    if (parentDir == dir) return null;
+    dir = parentDir;
+  }
+  return dir;
+}
+
+bool _isPolymerElement(String element) {
+  return element.contains('-') && (element.toLowerCase() == element);
+}
+
+String _toCamelCase(String s) {
+  return s[0].toUpperCase() + s.substring(1);
+}
+
+void _createBoilerPlate(String element, String directory,
+                        int distanceToPackageRoot) {
+  var segments = element.split('-');
+  var capitalizedName = segments.map((e) => _toCamelCase(e)).join('');
+  var underscoreName = element.replaceAll('-', '_');
+
+  String html = '''
+<dom-module id="$element">
+    <template>
+        <style>
+	   /* CSS rules for your element */
+	</style>
+	
+      <!-- local DOM for your element -->
+      
+    </template>
+</dom-module>
+''';
+
+  String htmlFile = path.join(directory, underscoreName + '.html');
+  new File(htmlFile).writeAsStringSync(html);
+
+  String dart = '''
+@HtmlImport('$underscoreName.html')
+import 'package:polymer/polymer.dart';
+import 'package:web_components/web_components.dart' show HtmlImport;
+
+/**
+ * A Polymer $element element.
+ */
+@PolymerRegister('$element')
+class $capitalizedName extends PolymerElement {
+
+  /// Constructor used to create instance of ${capitalizedName}.
+  ${capitalizedName}.created() : super.created();
+
+  /*
+   * Optional lifecycle methods - uncomment if needed.
+   *
+
+  /// Called when an instance of $element is inserted into the DOM.
+  attached() {
```

Comment:
Probably best to keep the super call even though it doesn't do anything today? dart2js should just eliminate it since its dead code and it has a higher chance of not breaking in the future.